File: src/main/java/com/nugurang/dao/ImageDaoImpl.java
package com.nugurang.dao;

import static com.nugurang.entity.QImageEntity.imageEntity;
import static com.nugurang.entity.QXrefArticleImageEntity.xrefArticleImageEntity;
import com.nugurang.entity.ImageEntity;
import com.querydsl.jpa.impl.JPAQueryFactory;
import java.util.List;
import org.springframework.stereotype.Repository;

@Repository
public class ImageDaoImpl implements ImageDaoCustom {
    private final JPAQueryFactory queryFactory;

    public List<ImageEntity> findAllByArticleId(Long article) {
        return queryFactory.selectFrom(imageEntity).innerJoin(xrefArticleImageEntity).on(imageEntity.id.eq(xrefArticleImageEntity.image.id)).where(xrefArticleImageEntity.article.id.eq(article)).fetch();
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public ImageDaoImpl(final JPAQueryFactory queryFactory) {
        this.queryFactory = queryFactory;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/dao/ImageDaoImpl.kt
package com.nugurang.dao

import com.nugurang.entity.ImageEntity
import com.nugurang.entity.QImageEntity
import com.nugurang.entity.QXrefArticleImageEntity
import com.querydsl.jpa.impl.JPAQueryFactory
import org.springframework.stereotype.Repository

@Repository
class ImageDaoImpl(private val queryFactory: JPAQueryFactory) : ImageDaoCustom {
    override fun findAllByArticleId(article: Long): List<ImageEntity> {
        return queryFactory.selectFrom(QImageEntity.imageEntity)
            .innerJoin(QXrefArticleImageEntity.xrefArticleImageEntity)
            .on(QImageEntity.imageEntity.id.eq(QXrefArticleImageEntity.xrefArticleImageEntity.image.id))
            .where(QXrefArticleImageEntity.xrefArticleImageEntity.article.id.eq(article)).fetch()
    }
}